File: week6/sentimental-credit/credit.py
# TODO
from cs50 import get_int

number = get_int("Number: ")
# Make a copy of the number inputed by the user
copy = number

# Convert the integer to a string
string = str(number)

# Count the number of digits in the integer
length = len(string)


def main():
    # Check if the card number is valid
    sum = is_valid(string, copy)

    # If sum isn't diviible by 10, then print "INVALID"
    if sum % 10 != 0:
        print("INVALID")

    # Check if it is VISA
    if length == 13 or length == 16:
        quotient = int(number / 10 ** (length - 1))
        if quotient == 4:
            print("VISA")
            return 0
    # Check if it is American Express
    elif length == 15:
        quotient = int(number / 10 ** (length - 2))
        if quotient == 34 or quotient == 37:
            print("AMEX")
            return 0
    # Check if it is Master Card
    if length == 16:
        quotient = int(number / 10 ** (length - 2))
        if quotient >= 51 and quotient <= 55:
            print("MASTERCARD")
            return 0
    # If none of the above is true, then print "INVALID"
    else:
        print("INVALID")


# Checks if the card number is valid
def is_valid(string, copy):
    sum = 0
    for i in range(len(string)):
        # Assigns
        if i % 2 == 0:
            i = copy % 10
        else:
            i = 2 * (copy % 10)
            if i > 9:
                i = int(i % 10) + int(i / 10)
        sum += i
        copy = int(copy / 10)
    return sum


main()

File: week1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

// Counter
int i = 1;

// Sum
int sum = 0;


int main(void)
{
    // Prompts the user for a number
    long number = get_long("Number: ");
    long str = number;
    do
    {
        int j = 2 * (number % 10);

        // Multiplies the digits by 2
        if (i % 2 == 0)
        {

            //Adds the digit to the sum
            sum = sum + (j % 10) + (j / 10);
            number = number / 10;
            i++;
        }

        else
        {
            sum = sum + (number % 10);
            number = number / 10;
            i++;
        }
    }

    // Checks the condition
    while (number != 0);

    // Checks if it is valid
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
    }

    else
    {
        // Check if it is MasterCard or Visa
        if (str > (1e15))
        {
            if (str / (1e15) >= 4 && str / (1e15) < 5)
            {
                printf("VISA\n");
            }

            else if (str / (1e14) >= 51 && str / (1e14) < 56)
            {

                printf("MASTERCARD\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }

        // Checks if it is American Express
        else if (str > 1e14)
        {
            if (str / (1e13) >= 34 && str / (1e13) < 35)
            {
                printf("AMEX\n");
            }

            else if (str / (1e13) >= 37 && str / (1e13) < 38)
            {
                printf("AMEX\n");
            }

            else
            {
                printf("INVALID\n");
            }
        }

        //Checks if it is Visa
        else if (str > (1e12))
        {
            if (str / (1e12) >= 4  && str / (1e12) < 5)
            {
                printf("VISA\n");

            }
            else
            {
                printf("INVALID\n");
            }
        }

        else
        {
            printf("INVALID\n");

        }
    }

}